RadixWallet/Features/AssetTransferFeature/Components/TransferAccountList/TransferAccountList+Reducer.swift

```diff
@@ -290,28 +286,43 @@ extension TransferAccountList {
 		return .none
 	}
 
-	private func determineAdditionalRequiredSignatures(
-		_ receivingAccount: ReceivingAccount.State,
-		forAssets assets: IdentifiedArrayOf<ResourceAsset.State>
-	) -> Effect<Action> {
-		if case let .profileAccount(value: userOwnedAccount) = receivingAccount.recipient {
+	private func signaturesStatusEffect(_ state: State, receivingAccountId: ReceivingAccount.State.ID) -> Effect<Action> {
+		guard
+			let receivingAccount = state.receivingAccounts[id: receivingAccountId],
+			let recipient = receivingAccount.recipient
+		else {
+			return .none
+		}
+
+		typealias DepositStatus = ResourceAsset.State.DepositStatus
+		switch recipient {
```

Comment:
Both cases, `profileAccount` and `externalAccount` do have in their essence the same logic, the difference is how the deposit statuses are determined, maybe have something like this:
```swift
let resourceAddresses = receivingAccount.assets.map(\.id)
if !resourceAddresses.isEmpty {
  await send(.internal(.setAllDepositStatus(accountId: receivingAccountId, status: .loading)))
  let depositStatuses: Dictionary = try await {
     switch recipient {
       case let .profileAccount(account):
           return try await getStatusesForProfileAccount(...)
       case let .addressOfExternalAccount(account):
           return try await getStatusesForExternalAccount(...)
     }
  }()
  
 await send(.internal(.setDepositStatus(accountId: receivingAccountId, values: values)))
}
```